### PR TITLE
LOK-2657: BFF work for adding a user

### PR DESCRIPTION
**Pull request by Shahbaz-dataq**
## Description
User addUser grapgQL mutation for adding a user.

mutation AddUser {    
  addUser(user: {username: "shahbaz", email: "$shahbaz@opennms", firstName: "Shahbaz", enabled: true, credentials: {value: "admin"}})
}

create a mapper to map client user on UserRepresentation in BFF.

public class UserRepresentation {
    private String self;
    private String id;
    private String origin;
    private Long createdTimestamp;
    private String username;
    // User must always be enabled by default
    private boolean enabled = true;
    private String firstName;
    private String lastName;
    private String email;
    private List<String> requiredActions = new ArrayList<>();
    private List<CredentialRepresentation> credentials = new ArrayList<>();
}

public class CredentialRepresentation {
    private String id;
    // Type can be fixed as password for now until we support other mechanisms
    private String type = "password";
    private String userLabel;
    private Long createdDate;
    private String value;
    protected Boolean temporary;
}

## Jira link(s)
- https://opennms.atlassian.net/browse/LOK-2657

### CODE REVIEW

File: ui/src/store/Views/userStore.ts [hunk 1]
```diff
@@ -49,43 +55,73 @@ export const useUserStore = defineStore('userStore', {
...
     },
-    saveUser() {
+    async saveUser() {
+      this.validationErrors = {}
       if (!this.selectedUser || !this.validateUser(this.selectedUser)) {
         return
       }
+      if (this.checkForDuplicateUsernameOrEmail()?.email === this.selectedUser.email) {
+        showSnackbar({
+          msg: 'User exits with same email address.',
```
Comment (synqotik) [inline ui/src/store/Views/userStore.ts:99]: Should be `exists` instead of `exits`

File: ui/src/store/Views/userStore.ts [hunk 2]
```diff
@@ -49,43 +55,73 @@ export const useUserStore = defineStore('userStore', {
...
+          error: true
+        })
+        return
+      }
+      if (this.checkForDuplicateUsernameOrEmail()?.username === this.selectedUser.username) {
+        showSnackbar({
+          msg: 'User exits with same username.',
```
Comment (synqotik) [inline ui/src/store/Views/userStore.ts:106]: Should be `exists` instead of `exits`

File: ui/src/store/Views/userStore.ts [hunk 3]
```diff
@@ -49,43 +55,73 @@ export const useUserStore = defineStore('userStore', {
...
-      this.userEditMode = CreateEditMode.None
-      this.isModalVisible = false
+      await createNewUser({user: userInput})
 
+      if (!errorWhileCreatingUser.value) {
+        this.closeModalHandler()
+        await this.getUsersList()
+      } else {
+        showSnackbar({
+          msg: 'Unexpected error occur while creating User.',
```
Comment (synqotik) [inline ui/src/store/Views/userStore.ts:121]: `occur` -> `occurred`

**Review by synqotik — APPROVED** [review]
Looks good! Just a couple small things to fix before merging.

**Review by faizanjanjua — COMMENTED** [review]
Looks Good to me

**Pull request merged**